### DISCUSSION TITLE
Usage with pgxpool

### DISCUSSION THREAD

**Discussion by KasonBraley** [XSAM/otelsql#148] [Q&A] [answered]
Is it possible to use this with pgxpool from the [pgx](https://github.com/jackc/pgx) library? I'm currently using https://github.com/exaring/otelpgx to get traces & metrics from pgxpool, and this pkg for my other database connections. But if I could consolidate dependencies and just use this one that would be great. I see some issues related to pgx, but haven't seen anything related to pgxpool. Just figured I would ask before going down a rabbit hole of trying to figure out if it works.

**Comment by XSAM** [comment, marked as answer]
> Is it possible to use this with [pgx](https://github.com/jackc/pgx)pool from the pgx library?

Per pgx's documents, it includes an adapter for the standard `database/sql interface`. So, it is possible to use this with `pgx`.

But `pgx` also mentions some features are unavailable with `dashbase/sql`. https://github.com/jackc/pgx#choosing-between-the-pgx-and-databasesql-interfaces. So, I guess whether to use this library depends on your use cases of the `pgx`.

**Reply by KasonBraley** [reply]
I appreciate the quick response, thank you.

**Comment by mfridman** [comment]
Is there an example on how one might configure this?

For example, many code bases would have something like:

```go
pool, err := pgxpool.NewWithConfig(ctx, config)
if err != nil {
	return nil, err
}
```

Which returns a `*pgxpool.Pool`, how does one add this otelsql package?

EDIT: Ahh, I think I missed your point, the idea is we'd convert the `*pgxpool.Pool` to a `*sql.DB` via the adapter:

```
db := stdlib.OpenDBFromPool(pool)
```
